Mimir/Util/StateGetter.cs

```diff
@@ -119,29 +127,44 @@ public class StateGetter(IStateService stateService)
             _ => null,
         };
     }
-
-    public async Task<List<RuneState>> GetRuneStatesAsync(Address avatarAddress)
+    
+    public async Task<AllRuneState> GetRuneStatesAsync(Address avatarAddress)
     {
-        var state = await stateService.GetState(
-            RuneSlotState.DeriveAddress(avatarAddress, BattleType.Arena));
-        var runeSlotState = state switch
+        var serialized = await GetStateAsync(avatarAddress, Addresses.RuneState);
+        AllRuneState allRuneState;
+        if (serialized is null)
```

Comment:
What if we reverse the `if` condition with explicit type?
```suggestion
        if (serialized is List list)
```

Reply:
I amended it in abb0fa287062685d87a7f94ba09f179fbe1f898c 🙏🏻